digEmailExtractor/EE.py

```diff
@@ -0,0 +1,301 @@
+        # print "\n"
+        return clean_results
+
+    def normalize(self, clean, unclean, output_format):
+        if self.output_format == EE.EE_OUTPUT_FORMAT_LIST:
+            return clean
+        else:
+            output = []
+            for co in clean:
+                email = {}
+                email['email'] = co
+                tmp_unclean = list(unclean)
+                # print co, tmp_unclean
+                if co in [username.strip() + "@" + domain.strip() for username, domain in tmp_unclean if domain and username]:
+                    # print co
+                    for tuc in tmp_unclean:
+                        username, domain = tuc
+                        tuc_string = username.strip() + "@" + domain.strip()
+                        if tuc_string == co:
+                            unclean.remove(tuc)
+                            continue
+                        domain = self.clean_domain(domain)
+                        username = EE.clean_username(username)
+                        if domain and username:
+                            email_string = username + "@" + domain
+                            if email_string == co:
+                                email['obfuscation'] = 'True'
+                    if 'obfuscation' not in email:
+                        email['obfuscation'] = 'False'
+                else:
+                    email['obfuscation'] = 'True'
+                output.append(email)
+            return output
+
+
+
+
+
+
+
+
+
+            # uc_tc = self.clean(unclean)
+            # for co in clean:
+            #     email = {}
+            #     email['email'] = co
+            #     if co in unclean:
+            #         for uo in uc_tc:
+            #             if co == uo:
+            #                 email['obfuscation'] = 'True'
+            #                 break
+            #         if obfuscation not in email:
+            #             email['obfuscation'] = 'False'
+            #     else:
+            #         email['obfuscation'] = 'True'
+            #     output.append(email)
+            return output
+
+    def extract_email(self, string, return_as_string=False):
+        """Extract email address from string.
+        :param string: the text to extract from
+        :param return_as_string: whether to return the result as a string of comma-separated values or
+        as a set
+        :type return_as_string: Boolean
+        """
+        line = string.lower().replace('\n', ' ').replace('\r', '')
+        line = re.sub(r"[*?]+", " ", line)
+        line = re.sub(r"\\n", " ", line)
+        line = re.sub(r"\s+g\s+mail\s+", " gmail ", line)
+        # print line
+        # return EE.extract_domain(line)
+
+        matches = re.findall(self.email_regex, line)
+        clean_results = self.clean(matches)
+        # matches = [username.strip() + "@" + domain.strip() for username, domain in matches if domain and username]
```

Comment:
Commented out code should be removed or captured in an if/else block if the functionality is needed.